### PR TITLE
Created the draft structure of the notebook

### PR DESCRIPTION
**Pull request by TolgoAI**
Hi Max! I have created a draft structure of the notebook for the multiclass classifier to predict feet items. 
Please have a look.

### CODE REVIEW

**Review by maximilian-hildebrandt — APPROVED** [review]
Overall Feedback
- Overall structure is good, you covered the main sections that are required to achieve the technical objectives.
- Comments (where existing) are insightful and describe the relevant objectives per section.
- Model training & evaluation lack some depth, needs a little bit more work. My other comments cover mostly minor details or learnings based on experience and/or reading literature.

Exploratory data analysis
- First conduct exploratory data analysis, then process data, since you need to understand the unique properties of the data first to be able to decide on the right processing steps.
- Be more specific in regard to the visualizations:
- Create a bar chart showing the class distribution of the independent variable (x axis distinct items, y axis frequency)
- Create a box plot for input features and histograms/kernel density plots for each variable
- Compute a correlation matrix (be mindful of the type of correlation and the scale level of the variable, pearson vs. spearman correlation, for example)
- Avoid 3d visualizations since mostly hard to consume.

Pre-processing

- Create a train/validation/test split (e.g., 50%, 25%, 25% according to Hastie et al. (2009, p. 222). The 3rd dataset is important to test the final performance. You never want to touch this dataset until you do the final test on the fine-tuned model. Once you used it, it loses its "value". 
- Comments on structure: Train-test split creation can be considered pre-processing of the data, same for classifier object creation. After pre-processing, the next major section should be "Model training".

Model training
- Please create relevant sections for the ML algorithms you would like to use, including any baseline classifier (e.g., dummy classifier). Minimum requirement: 
- 1 ML algorithm (suggestion to try out 2-3 to see different behaviors)
- 1 rule-based algorithm (from your other file after finalization of that script) for comparison
- Add a section for hyperparameter optimization: Which parameters to optimize? How? Which values would you like to try out?

Model evaluation

- Add comments to this section, objectives or rationales not yet clear here
- Estimating the accuracy of the classifier is already part of the evaluation, re-structure accordingly, same for new object classification (on the "test set" as defined by Hastie et al. above)
- Create a section for k-fold cross-validation
- "# Creating a dataset with imbalanced binary classes:  # Negative class (0) is 'not digit 1' # Positive class (1) is 'digit 1'" not clear why you inserted this here. Also, in multiclass classification you would have more than two classes. This should be part of pre-processing.

Deployment
- Add a section to output predictions for the digital product.
- For the beginning, save the model as a file, and create an .xls file with the testset (as defined by Hastie et al., 2009) and corresponding predictions in an additional column
- Details to be aligned with WD team (e.g., Flask interface).

Nitty-gritty detail suggestions (optional)
- Add a comment field in the beginning after the heading re-stating the user problem in one sentence & the technical objective (you cannot over-communicate this in documents)
- Add a comment with the structure of headings in the beginning, enabling a transparent overview of the full structure of the jupyter notebook
- Use numerals to number the level 1 headings, otherwise hard to digest
- Use the same grammatical structure in comments (either with -ing or without it, reviewing vs. review)
- Keep headings short (e.g. "data loading") and use full sentences in the comments (e.g. "train the classifier (fit the estimator) using the training data")
